.github/workflows/windows.yml

```diff
@@ -77,36 +77,75 @@ jobs:
         echo "MKLROOT=${{ github.workspace }}\mkl" | Out-File -FilePath $env:GITHUB_ENV  -Encoding utf8 -Append
       shell: powershell
 
-    - name: Prepare vcpkg
-      uses: lukka/run-vcpkg@v7.4
-      with:
-        vcpkgArguments: protobuf pcre2 --overlay-ports="${{ github.workspace }}\vcpkg-override\ports\pcre2"
-        vcpkgGitCommitId: 8dddc6c899ce6fdbeab38b525a31e7f23cb2d5bb
-        vcpkgDirectory: ${{ github.workspace }}/vcpkg/
-        vcpkgTriplet: x64-windows-static
-
-    # Windows CPU only minimal build
-    - name: Build Release # @TODO this is actually a debug build until the ninja generator gets fixed
-      uses: lukka/run-cmake@v3
+    - name: Cache vcpkg
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-vcpkg
       with:
-        buildDirectory: ${{ github.workspace }}/build
-        cmakeAppendedArgs: '-G Ninja
-          -DCMAKE_BUILD_TYPE="Release"
-          -DUSE_WASM_COMPATIBLE_SOURCE="OFF"
-          -DUSE_STATIC_LIBS="TRUE" 
-          -DCMAKE_CXX_COMPILER_LAUNCHER=${{github.workspace}}\ccache.exe
-          -DCMAKE_C_COMPILER_LAUNCHER=${{github.workspace}}\ccache.exe
-        '
-        cmakeListsOrSettingsJson: CMakeListsTxtAdvanced
-        cmakeListsTxtPath: ${{ github.workspace }}/CMakeLists.txt
-        useVcpkgToolchainFile: true
-        cmakeBuildType: Release
+
+        path: C:\vcpkg\installed # cache only the latest installed packages
```

Comment:
The cache is prone to breakages and it's not proper. I'd remove it, you're just compiling one very tiny package anyways.